tests/integration_tests/functional/test_config_space.py

```diff
@@ -0,0 +1,351 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Tests on devices config space."""
+
+import json
+import os
+import random
+import re
+import string
+from threading import Thread
+import subprocess
+import platform
+
+import host_tools.logging as log_tools
+import host_tools.network as net_tools  # pylint: disable=import-error
+
+# pylint: disable=global-statement
+PAYLOAD_DATA_SIZE = 20
+
+
+def test_net_change_mac_address(test_microvm_with_ssh, network_config,
+                                change_config_space_bin):
+    """Test changing the MAC address of the network device."""
+    global PAYLOAD_DATA_SIZE
+
+    test_microvm = test_microvm_with_ssh
+    test_microvm.spawn()
+    test_microvm.basic_config(boot_args="ipv6.disable=1")
+
+    # Data exchange interface ('eth0' in guest).
```

Comment:
does it worth adding a tests with 2 block devices and 2 network interfaces. The block devices added first?

Reply:
At the moment, on both x86_64 and aarch64, the test microvm contains one blk device and two net devices. The side effect of adding one more blk device would be the shift by one page in memory mappings for the net devices.  If we target the logic which determines the start address of a network device memory mapping, it is idempotent to this change because the start address is not dependent on any order of the memory mappings (it is determined based on the interface name - which is determined in relation with the configured guest IPv4 address of the network interface of interest). I think that adding one more blk device to the test microvm does not bring a side effect which might change the behavior of the test.